src/apps/bristleback_apps/serial_payload_example/user_code/user_code.cpp

```diff
@@ -36,44 +62,94 @@ void setup(void) {
   vTaskDelay(pdMS_TO_TICKS(5));
   // enable Vout, 12V by default.
   IOWrite(&BB_PL_BUCK_EN, 0);
+
+  // turn off LEDs
+  IOWrite(&LED_BLUE, 0);
+  IOWrite(&LED_GREEN, 0);
+  IOWrite(&LED_RED, 0);
 }
 
 void loop(void) {
   /* USER LOOP CODE GOES HERE */
-
-  // Read a byte if it is available
+  // Read a cluster of Rx bytes if available
+  // -- A timer is used to try to keep clusters of bytes (say from lines) in the same output.
+  static int64_t readingBytesTimer = -1;
+  // Note - PLUART::setUseByteStreamBuffer must be set true in setup to enable bytes.
+  if (readingBytesTimer == -1 && PLUART::byteAvailable()) {
+    // Get the RTC if available
+    RTCTimeAndDate_t time_and_date = {};
+    rtcGet(&time_and_date);
+    char rtcTimeBuffer[32];
+    rtcPrint(rtcTimeBuffer, &time_and_date);
+    printf("[payload-bytes] | tick: %llu, rtc: %s, bytes:", uptimeGetMs(), rtcTimeBuffer);
+    // not very readable, but it's a compact trick to overload our timer variable with a -1 flag
+    readingBytesTimer = (int64_t)((u_int32_t)uptimeGetMs());
```

Comment:
why not make the `readingBytesTimer` a uint64_t, initialize it to 0 and make all the compares against 0? That would remove the need for the casting magic.

Is there an edge case with 0 I'm not seeing?

Reply:
Seconded, it took me a second to realize what was going on. I think it would make it more readable if it was kept at a `uint64_t`. That way we also would risk truncating `uptimeGetMs` (although very unlikely to ever happen).

Reply:
@towynlin @victorsowa12 @matt001k - so there's actually an evil quirk with `uptimeGetMs` (our function, our fault) that bit us during some very early stm32 work - I know Russ was in the weeds on that one, but I don't know if you remember it Victor? If RTC is not set, the return value will actually rollover at uint32_max. If RTC is set, it will rollover at uint64_max.

So the very slim edge case is a rollover at ~32 days of runtime falsely triggering a new byte cluster while we're already in the middle of one. 🤷

Reply:
Sounds familiar. Adding this for context for the SC-208149 ticket. Took a quick look through and I see that `uptimeGetMs` refers to FreeRTOS ticks when there is no RTC. FreeRTOS ticks are stored as a `uint32_t`, so it will rollover.

I think in the context of the system you would need three things to get this rollover:
1. no gps signal for 32 days (or an RTC-0 blocking node ahead of you in the chain)
2. power controller enabled continuously
3. power controller tick timing enabled

I don't know the use case for PME so I don't know if this is a likely scenario. 🤷‍♂️